servicetest/conftest.py

```diff
@@ -59,14 +59,35 @@ def agent(request):
     # and thus the path should be defined on the module level.
     logfile_path = request.module.logfile_path()
 
+    # TODO: Remove this "caps dirs" stuff when done with service manifest support?
     # Check if caps_dirs is defined
     if 'caps_dirs' in request.module.__dict__:
         caps_dir, default_caps_dir = request.module.caps_dirs()
     else:
         caps_dir = None
         default_caps_dir = None
 
-    agent_handler = SoftwareContainerAgentHandler(logfile_path, caps_dir, default_caps_dir)
+    standard_location = None
+    default_location = None
+
+    if "service_manifests" in request.module.__dict__:
+        manifests = request.module.service_manifests()
+        for manifest in manifests:
+            if manifest.is_default():
+                default_location = manifest.location() + "/service-manifest.default.d/"
+                if not os.path.exists(default_location):
+                    os.makedirs(default_location)
+                service_manifest_file = default_location + manifest.name()
+            else:
+                standard_location = manifest.location() + "/service-manifest.d/"
+                if not os.path.exists(standard_location):
+                    os.makedirs(standard_location)
+                service_manifest_file = standard_location + manifest.name()
+
+            with open(service_manifest_file, "w") as fh:
+                fh.write(manifest.json_as_string())
```

Comment:
This assumes that the file is always created with user read permissions set. Maybe we should make sure they are set?